freeipa/src/main/java/com/sequenceiq/freeipa/service/image/FreeIpaImageProvider.java

```diff
@@ -79,82 +77,79 @@ public List<ImageWrapper> getImages(ImageSettingsRequest imageSettings, String r
                     .collect(Collectors.toList());
         } else {
-            return filterImages(images, imageOs, platform, region).stream()
-                    .filter(image -> image.getImageSetsByProvider().containsKey(platform))
-                    .collect(Collectors.toList());
+            return filterImages(images, imageFilterSettings);
         }
     }
 
-    private Boolean isMatchingImageIdInRegion(String imageId, String region, String platform, Image img) {
-        return Optional.ofNullable(img.getImageSetsByProvider().get(platform).get(region))
-                .map(reg -> reg.equalsIgnoreCase(imageId))
+    private Boolean isMatchingImageIdInRegion(FreeIpaImageFilterSettings imageFilterSettings, Image img) {
+        return Optional.ofNullable(img.getImageSetsByProvider().get(imageFilterSettings.getPlatform()).get(imageFilterSettings.getRegion()))
+                .map(reg -> reg.equalsIgnoreCase(imageFilterSettings.getImageId()))
                 .orElse(false);
     }
 
     private boolean hasSameUuid(String imageId, Image img) {
         return img.getUuid().equalsIgnoreCase(imageId);
     }
 
-    private boolean filterRegion(String region, String platform, Image img) {
-        Set<String> regionSet = img.getImageSetsByProvider().get(platform).keySet();
-        return CollectionUtils.containsAny(regionSet, region, DEFAULT_REGION);
-    }
-
-    private List<Image> filterImages(List<Image> imageList, String os, String platform, String region) {
-        Predicate<Image> predicate = img -> img.getOs().equalsIgnoreCase(os) &&
-                img.getImageSetsByProvider().containsKey(platform) &&
-                filterRegion(region, platform, img);
-        Map<Boolean, List<Image>> partitionedImages =
-                Optional.ofNullable(imageList).
-                        orElse(Collections.emptyList()).
-                        stream()
-                        .collect(Collectors.partitioningBy(predicate));
-        if (hasFiltered(partitionedImages)) {
-            LOGGER.debug("Used filter for: | {} | Images filtered: {}",
-                    os,
-                    partitionedImages.get(false).stream().map(Image::toString).collect(Collectors.joining(", ")));
-            List<Image> images = partitionedImages.get(true);
-            return providerSpecificImageFilter.filterImages(platform, images);
+    private boolean filterRegion(FreeIpaImageFilterSettings imageFilterSettings, Image img) {
+        Set<String> regionSet = img.getImageSetsByProvider().get(imageFilterSettings.getPlatform()).keySet();
+        return CollectionUtils.containsAny(regionSet, imageFilterSettings.getRegion(), DEFAULT_REGION);
+    }
+
+    private List<Image> filterImages(List<Image> candidateImages, FreeIpaImageFilterSettings imageFilterSettings) {
+        LOGGER.debug("Filtering images with the following parameters: {}", imageFilterSettings);
```

Comment:
`FreeIpaImageFilterSettings` doesn't have toString implementation. Either it should be implemented, or it should be converted to record